compiler/rustc_passes/src/check_attr.rs

```diff
@@ -900,6 +900,31 @@ impl Visitor<'tcx> for CheckAttrVisitor<'tcx> {
         intravisit::walk_trait_item(self, trait_item)
     }
 
+    fn visit_struct_field(&mut self, struct_field: &'tcx hir::StructField<'tcx>) {
+        self.check_attributes(
+            struct_field.hir_id,
+            &struct_field.attrs,
+            &struct_field.span,
+            Target::Field,
+            None,
+        );
+        intravisit::walk_struct_field(self, struct_field);
+    }
+
+    fn visit_arm(&mut self, arm: &'tcx hir::Arm<'tcx>) {
+        self.check_attributes(arm.hir_id, &arm.attrs, &arm.span, Target::Arm, None);
+        intravisit::walk_arm(self, arm);
+    }
+
+    fn visit_macro_def(&mut self, macro_def: &'tcx hir::MacroDef<'tcx>) {
+        for attr in macro_def.attrs {
+            if self.tcx.sess.check_name(attr, sym::inline) {
+                self.check_inline(macro_def.hir_id, attr, &macro_def.span, Target::MacroDef);
+            }
+        }
```

Comment:
Why aren't you using `check_attributes` here?

Reply:
To elaborate more on my question: If there is a reason, we should document it or look into whether we can fix any issues with it. This doesn't have to happen in this PR, so leaving a `// FIXME: explain reason and possible fix here` on it is totally alright.

Reply:
This is the part I missed just focusing on fixing, and I soon added a commit.

Reply:
There are two problems.
First, I don't know how to turn non_exported_macro_attrs into warning.
Second, the flow can access from check_inline to Field, Arm, and MacroDef through check_attributes, but finally, there is no warning message from the test (attr-usage-inline).

I kept trying, but it failed and I don't know what to do. Could I get a hint?